tests/pytests/unit/utils/test_network.py

```diff
@@ -1,8 +1,1560 @@
+    ):
+        assert network.generate_minion_id() == "localhost"
+
+
+def test_generate_minion_id_platform_fqdn():
+    """
+    Test if fqdn is picked up.
+    """
+    with patch("platform.node", MagicMock(return_value="localhost")), patch(
+        "socket.gethostname", MagicMock(return_value="ip6-loopback")
+    ), patch("socket.getfqdn", MagicMock(return_value="pick.me")), patch(
+        "socket.getaddrinfo",
+        MagicMock(return_value=[(2, 3, 0, "localhost", ("127.0.1.1", 0))]),
+    ), patch(
+        "salt.utils.files.fopen", mock_open()
+    ), patch(
+        "salt.utils.network.ip_addrs",
+        MagicMock(return_value=["127.0.0.1", "::1", "fe00::0", "fe02::1"]),
+    ):
+        assert network.generate_minion_id() == "pick.me"
+
+
+def test_generate_minion_id_platform_localhost_addrinfo():
+    """
+    Test if addinfo is picked up.
+    """
+    with patch("platform.node", MagicMock(return_value="localhost")), patch(
+        "socket.gethostname", MagicMock(return_value="ip6-loopback")
+    ), patch("socket.getfqdn", MagicMock(return_value="ip6-localhost")), patch(
+        "socket.getaddrinfo",
+        MagicMock(return_value=[(2, 3, 0, "pick.me", ("127.0.1.1", 0))]),
+    ), patch(
+        "salt.utils.files.fopen", mock_open()
+    ), patch(
+        "salt.utils.network.ip_addrs",
+        MagicMock(return_value=["127.0.0.1", "::1", "fe00::0", "fe02::1"]),
+    ):
+        assert network.generate_minion_id() == "pick.me"
+
+
+def test_generate_minion_id_platform_ip_addr_only():
+    """
+    Test if IP address is the only what is used as a Minion ID in case no DNS name.
+    """
+    with patch("platform.node", MagicMock(return_value="localhost")), patch(
+        "socket.gethostname", MagicMock(return_value="ip6-loopback")
+    ), patch("socket.getfqdn", MagicMock(return_value="ip6-localhost")), patch(
+        "socket.getaddrinfo",
+        MagicMock(return_value=[(2, 3, 0, "localhost", ("127.0.1.1", 0))]),
+    ), patch(
+        "salt.utils.files.fopen", mock_open()
+    ), patch(
+        "salt.utils.network.ip_addrs",
+        MagicMock(return_value=["127.0.0.1", "::1", "fe00::0", "fe02::1", "1.2.3.4"]),
+    ):
+        assert network.generate_minion_id() == "1.2.3.4"
+
+
+def test_gen_mac():
+    with patch("random.randint", return_value=1) as random_mock:
+        assert random_mock.return_value == 1
+        ret = network.gen_mac("00:16:3E")
+        expected_mac = "00:16:3E:01:01:01"
+        assert ret == expected_mac
+
+
+@pytest.mark.parametrize(
+    "mac_addr",
+    (
+        ("31337"),
+        ("0001020304056"),
+        ("00:01:02:03:04:056"),
+        ("a0:b0:c0:d0:e0:fg"),
+    ),
+)
+def test_mac_str_to_bytes_exceptions(mac_addr):
+    pytest.raises(ValueError, network.mac_str_to_bytes, mac_addr)
+
+
+def test_mac_str_to_bytes():
+    assert b"\x10\x08\x06\x04\x02\x00" == network.mac_str_to_bytes("100806040200")
+    assert b"\xf8\xe7\xd6\xc5\xb4\xa3" == network.mac_str_to_bytes("f8e7d6c5b4a3")
```

Comment:
See https://gist.github.com/s0undt3ch/92d065815f4ce526345f72d40a40074d#file-pytest-output-txt as to why

Reply:
Don't need the gist, this has been best practice in C/C++ since the turn of the century
condition == fixed static check
Too many dev's do it the other way, blame original K&R for that bad example, but then that was the early days :smile: , done

Reply:
Also leaving test_in_subnet alone, since got that caplog and it is so short, not worth parameterizing